### PR TITLE
Implement documentation index

### LINKED ISSUE #6: Missing implementation of index.md
As the tutorial notebook states, we should implement the index markdown file.
The content should be:

- Search Algorithm: a description of the search algorithm, and how the goodness of the equation was determined?
- Search Space: which search space was used and how was the search otherwise constrained?

### PR DESCRIPTION
**Pull request by hannesvoss**
# Description

This commit is implementing the index markdown file. The purpose of this file is to explain the following:

- Search Algorithm: a description of the search algorithm, and how the goodness of the equation was determined?
- Search Space: which search space was used and how was the search otherwise constrained?

- resolves #6 

# Type of change

- **docs**: Documentation only changes

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1,24 +1,32 @@
 # autora-theorist-Avengers
 
-Explain what your contribution is doing here
+## Search Algorithm
 
-You can include inline mathematics by wrapping LaTeX formatted math in `$` symbols: `$(3 < 4)$` will render as:  $(3 < 4)$
+The search algorithm employed by the theorist involves fitting a quadratic model to the provided dataset using the `ParabolaRegression` class. The algorithm follows these steps:
```
Comment (Krishnendu-Bose) [inline docs/index.md]: Because this documentation section carries 4 points, I think it's better if we make this more detailed.

Maybe rephrase this sentence as : The search algorithm employed by our theorist involves fitting the provided dataset by using a quadratic function which takes the shape of a parabola. Hence, we have defined the class containing the parabolic function as "ParabolaRegression" class. The algorithm follows these steps:

File: docs/index.md [hunk 2]
```diff
@@ -1,24 +1,32 @@
...
+   - Quadratic terms (squares and cross-products of the features).
+3. **Coefficient Calculation**: The coefficients are calculated using the least squares method by solving the normal equation \( A \beta = y \).
+4. **Model Fitting**: The model is fitted to the data by storing the calculated coefficients.
 
-```
-$$
-E(\mathbf{v}, \mathbf{h}) = -\sum_{i,j}w_{ij}v_i h_j - \sum_i b_i v_i - \sum_j c_j h_j
-$$
-```
+The goodness of the equation is determined using statistical measures such as the coefficient of determination (R²) and mean squared error (MSE).
```
Comment (Krishnendu-Bose) [inline docs/index.md]: The goodness of the fit of the equation to the data is determined using the following statistical measures which are commonly used in regression:

     -   R² -> The R² value measures the amount of variance present in the values of the dependent variable that can or cannot be properly explained by the independent variable. A R² value of 0 means the independent variables in question cannot explain the variances present in the dependent variable, whereas a R² value of 1 means that the independent variable can fully account for the variances in the dependent variable.
     
     - Mean Squared Error (MSE) -> The MSE value denotes the mean of the squared error difference between the predicted output values or labels and the original output values or labels. In the case of the parabolic function used by our theorist for fitting the data, the MSE is calculated between the actual output data points and the data points present on the parabola curve calculated by our theorist. This MSE value needs to be minimized by finding optimal coefficient values for each of the input feature terms present in the final parabola equation.

File: docs/index.md [hunk 3]
```diff
@@ -1,24 +1,32 @@
...
+- **Log-Transformed Terms**: The natural logarithm of the original features.
+- **Quadratic Terms**: The squares and cross-products of the original features.
 
-See more information on writing mathematical expressions [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions). 
+The search is constrained by the structure of the design matrix, which ensures that all relevant terms are included without overfitting. The constraints include:
 
+- **Feature Selection**: All features are considered in their original, log-transformed, and quadratic forms.
+- **Equation Complexity**: The design matrix is constructed to include all necessary terms while avoiding unnecessary complexity.
 
+These constraints help in efficiently exploring the search space and identifying the best-fitting quadratic equation.
```
Comment (Krishnendu-Bose) [inline docs/index.md]: I think a bit more explanation is required as to how the constraints employed on the design matrix prevent overfitting of the relevant terms.

**Review by Krishnendu-Bose — DISMISSED** [review]
Please take a look at the different changes that I requested in different sections of the index.md in order to make it more elaborate and clear to understand.

File: docs/index.md (hunk 1 above)
Comment (hannesvoss) [reply docs/index.md]: True 👍🏼

**Review by AlexanderDitz — APPROVED** [review]
